cloud/shared/bin/lib/terraform.py

```diff
@@ -1,37 +1,118 @@
+        lock_id: str,
+        terraform_template_dir: Optional[str] = None,
+        initialize=True):
     if not terraform_template_dir:
         terraform_template_dir = config_loader.get_template_dir()
-    tf_vars_filename = config_loader.tfvars_filename
 
-    terraform_cmd = f'terraform -chdir={terraform_template_dir}'
+    if initialize:
+        perform_init(
+            config_loader, terraform_template_dir, False)  # upgrade = False
+
+    terraform_cmd = f'terraform -chdir={terraform_template_dir} force-unlock -force {lock_id}'
+    print(f" - Run {terraform_cmd}")
+    subprocess.check_call(shlex.split(terraform_cmd))
+
+
+def perform_init(
+        config_loader: ConfigLoader,
+        terraform_template_dir: Optional[str] = None,
+        upgrade: bool = True):
+    if not terraform_template_dir:
+        terraform_template_dir = config_loader.get_template_dir()
+
+    init_cmd = f'terraform -chdir={terraform_template_dir} init'
+    if upgrade:
+        init_cmd += ' -upgrade'
 
     if config_loader.use_local_backend:
-        print(' - Run terraform init -upgrade -reconfigure')
-        subprocess.check_call(
-            shlex.split(f'{terraform_cmd} init -upgrade -reconfigure'))
+        init_cmd += ' -reconfigure'
     else:
-        print(' - Run terraform init -upgrade')
-        init_cmd = f'{terraform_cmd} init -input=false -upgrade'
+        init_cmd += ' -input=false'
         # backend vars file can be absent when pre-terraform setup is running
         if os.path.exists(os.path.join(terraform_template_dir,
                                        config_loader.backend_vars_filename)):
             init_cmd += f' -backend-config={config_loader.backend_vars_filename}'
-        subprocess.check_call(shlex.split(init_cmd))
+    print(f" - Run {init_cmd}")
+    output, exit_code = capture_stderr(init_cmd)
+    if exit_code > 0:
+        is_tty = sys.stdin.isatty()
+        # This is AWS-specific, and should be modified when we have actual
+        # Azure deployments
+        if 'state data in S3 does not have the expected content' in output:
+            match = re.search(r'value: ([0-9a-f]{32})', output)
+            if match:
+                digest = match.group(match.lastindex)
+                if is_tty:
+                    answer = input(
+                        "Would you like to fix this by setting the correct digest value? Ensure that no other deployment processes are in progress. [Y/n] >"
+                    )
+                    if answer.lower() in ['y', 'yes', '']:
+                        aws = AwsCli(config_loader)
+                        aws.set_lock_table_digest_value(digest)
+                        perform_init(
+                            config_loader, terraform_template_dir, upgrade)
+                        return
+                print(
+                    f"To fix the above error, set the LOCK_TABLE_DIGEST_VALUE environment variable to \"{digest}\" and rerun this command."
```

Comment:
**suggestion:** You'll need to add that variable to env-var-docs, right? Also, what happens if they add that lock id and then forget to remove it?

Reply:
Ah got it, thanks! Maybe we can make the print statement more clear by saying something like:

`To fix the above error, rerun the command with LOCK_TABLE_DIGEST_VALUE=\"{digest}\" before it.`

Thoughts? Just so there isn't any confusion about where it belongs.